pennylane/templates/state_preparations/arbitrary_state_preparation.py

```diff
@@ -79,19 +62,53 @@ def ArbitraryStatePreparation(weights, wires):
 
         @qml.qnode(dev)
         def vqe(weights):
-            qml.ArbitraryStatePreparations(weights, wires=[0, 1, 2, 3])
+            qml.templates.ArbitraryStatePreparation(weights, wires=[0, 1, 2, 3])
 
             return qml.expval(qml.Hermitian(H, wires=[0, 1, 2, 3]))
 
+    The shape of the weights parameter can be computed as follows:
+
+    .. code-block:: python
+
+        shape = qml.templates.ArbitraryStatePreparation.shape(n_wires=3)
+
+
     Args:
-        weights (tensor_like): The angles of the Pauli word rotations, needs to have length :math:`2^(n+1) - 2`
+        weights (tensor_like): Angles of the Pauli word rotations. Needs to have length :math:`2^(n+1) - 2`
             where :math:`n` is the number of wires the template acts upon.
-        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers or strings, or
-            a Wires object.
+        wires (Iterable): wires that the template acts on
     """
 
-    wires = Wires(wires)
-    _preprocess(weights, wires)
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires, do_queue=True):
+
+        shape = qml.math.shape(weights)
+        if shape != (2 ** (len(wires) + 1) - 2,):
+            raise ValueError(
+                f"Weights tensor must be of shape {(2 ** (len(wires) + 1) - 2,)}; got {shape}."
+            )
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+
+    def expand(self):
+
+        with qml.tape.QuantumTape() as tape:
+            for i, pauli_word in enumerate(_state_preparation_pauli_words(len(self.wires))):
+                qml.PauliRot(self.parameters[0][i], pauli_word, wires=self.wires)
```

Comment:
The parameters of an `Operation` are by definition stored in a list. Even if we only have one parameter array, it will be the first element of the parameter list. 

I was wondering if it is clearer to define 
```
weights = self.parameters[0]
``` 
which would help coders?

Reply:
I think it's fine the way it is.  Most of the time I would just accept a detail like that, but I was being particular and curious.